### ISSUE TITLE
Graphs question

### ISSUE THREAD

**Issue by 41south** [poblabs/weewx-belchertown#143]
I've got yaxis_min and max points set for my barometer graph, thinking that if those points were exceeded that the graph would auto adjust. Apparently that's not the case and I ended up with the attached graph when the pressure dropped below my yaxis_min setting. Is there something I missed or do I need to allow a full yaxis scale for the barometer?

![Capture](https://user-images.githubusercontent.com/1725001/58451134-4c675b00-8166-11e9-8687-114c1e2b975b.JPG)

**Comment by poblabs** [comment]
I think you might want yaxis_softmax and yaxis_softmin? Try those out. Same
config method but if I recall they float the min max based on the values
rather than hard set the min max.

**Comment by poblabs** [comment]
Just checking if you were able to try this out?

**Comment by 41south** [comment]
Just waking up down here so I'll pop that in soon and see how it looks :)

**Comment by 41south** [comment]
I'm pretty sure that will fix it - thanks. Can you explain tick interval on the y axis in layman's terms? Does it help clean up the slightly odd looking y axis scale? I saw it discussed somewhere, maybe another one for the Wiki?

**Comment by poblabs** [comment]
The [tick interval](https://api.highcharts.com/highcharts/yAxis.tickInterval) is the interval in between the numbers on the y axis. On the barometer it was set to 0.01 from an old request. So that's why you see on my chart the numbers are so tightly grouped. See below... 29.920, 29.910, 29.000 - intervals of 0.01

![image](https://user-images.githubusercontent.com/3484775/58594944-147f1700-823d-11e9-94c6-5eaa49fea433.png)

I've got back and read my code and I seem to have a redundant setting in here because even though the user can specify `yaxis_tickinterval`, I'm still hardcoding the 0.01 on barometer. I think I'll remove that and just **instruct those who want the 0.01 interval to use `yaxis_tickinterval` instead**? Or leave the 0.01 default?

Anyways - if you change it to something like 0.1, you get the below:

```
        [[[barometer]]]
            color = "#BECC00"
            yaxis_tickinterval = 0.1
```

![image](https://user-images.githubusercontent.com/3484775/58595072-7b9ccb80-823d-11e9-92ec-ff98225e3b21.png)

**Comment by 41south** [comment]
Ah hah, yes that makes sense. I've just set mine to 5 as my readings are all hundreds and thousands being mbar .... it's just updated and 5 looks much better to me. I've got the softmin/max set a bit closer for testing to see how the graph reacts
As for the default, good question. The 0.01 seems way too much for hpa/mbar but possibly ok for your inHg.

**Comment by HoracioDos** [comment]
Hello!
I don't want to hijack this issue but I have another question about graphs. I see that ordinals go from N and back to N in wind speed and direction graph. Does weewx get wind direction from 0 to 259 or from 0 to 360?, and we have "two norths". Then I see this in skin.conf/weex.conf that seems to confirm the last option.
`graphs_windDir_ordinals = '{ 0: "N", 90: "E", 180: "S", 270: "O", 360: "N" }'`
It's a little weird to see N twice in the graph.

**Comment by HoracioDos** [comment]
Ok, I wanted to understand if it is due about how weewx or highcharts works. 
Thanks again!

**Comment by poblabs** [comment]
@HoracioDos You bring up a good question and I had this thought after replying to you. There must be a way to [categorize the wind on the y axis using yaxis categories](https://api.highcharts.com/highcharts/yAxis.categories) so that we only have 1 North. I will look into this.

**Comment by HoracioDos** [comment]
Hi
Other question. What happens with "Average Climatological Values for this year" graph if there is no rain in some months. I've connected the pluviometer a week ago and it didn't rain yet but all previous observations seem to be = 0 and not null or missing.
![AvgWeather2019](https://user-images.githubusercontent.com/4549674/58634707-5d2ce380-82c2-11e9-8a0f-5ec22ba4f0c7.png)

**Comment by poblabs** [comment]
It doesn't look like they are 0 or null since they show 25,05 and 9,4. 

Markers are disabled by default, and since there is only a single plot, and no markers, this gives the appearance that there is nothing there. 

What you could do is enable markers on the line charts. 

```
        [[[outTemp]]]
            zIndex = 1
            name = Max Temperature
            color = red
            [[[[marker]]]]
                enabled = true
        [[[outTemp_min]]]
            name = Min Temperature
            observation_type = outTemp
            aggregate_type = min
            color = "#7cb5ec"
            [[[[marker]]]]
                enabled = true
```

see `[[[[marker]]]]` enables the marker under the series. Save and run wee_reports and it should show the markers.

**Comment by HoracioDos** [comment]
Hello Pat
Rain totals seemed to be = 0. I'll try to enable markers. There are so many options!

**Comment by poblabs** [comment]
Yes, rain total is 0 you are right. But that doesn't affect the other plots based on that tooltip. It just looks like it is null because you only have 1 month of data and markers is disabled.

**Comment by poblabs** [comment]
Can you open a new issue with this information and attach your database?

**Comment by HoracioDos** [comment]
I think that I found the problem. My NOAA reports are old enough. At least 3 months but my database doesn't match. The oldest record in database is from May 11 2019 (I don't know what I did) Anyway,  as far as I understand, one month must be shown, but it shouldn't be Jan. I'll open a new issue.

**Comment by 41south** [comment]
Those soft limits work perfectly, closing this but another good one for the Wiki along with an explanation of the tick interval :)

**Comment by poblabs** [comment]
Great news! Yep added those to the wiki already.. the [soft limits are here](https://github.com/poblabs/weewx-belchertown/wiki/Belchertown-Charts-Documentation#yaxis_softmin-and-yaxis_softmax), and the [tick interval is here ](https://github.com/poblabs/weewx-belchertown/wiki/Belchertown-Charts-Documentation#yaxis_tickinterval) 😃

**Comment by HoracioDos** [comment]
Hello Pat.
Dew point is shown for Yaxis label in your temperature graph but there is name tag for temperature in graphs.conf. So why dewpoint is displayed?  Is it correct? I had the same the same graph until I added a second Y Axis for Humidity, then the first Y Axis for temperature fixed itself automagically.
Thanks!

**Comment by HoracioDos** [comment]
Hello!
I have another question. I found that aggregation type = max for [week] in graphs.conf.example. As far as I understand, temperature graph will plot max temperatures and not avg. Is that correct?
Thanks!

**Comment by poblabs** [comment]
@HoracioDos Correct, the highest aggregate_type will win. So if your week graph group has aggregate_type = max, [like you see here](https://github.com/poblabs/weewx-belchertown/blob/master/skins/Belchertown/graphs.conf.example#L144), then the full graph grouping will get the max. 

If you want to override that for an observation then [you can do it like this](https://github.com/poblabs/weewx-belchertown/blob/master/skins/Belchertown/graphs.conf.example#L188). You see how the barometer is under the week's "max" aggregate type - but we have ANOTHER aggregate_type beneath barometer which says avg. So every chart in this group will be max, but Barometer will be average because we override it. 

Does this help?

**Comment by HoracioDos** [comment]
Hello Pat.
> the closest aggregate_type to the observation will win

Yes, I've realized that, but my question is if it is correct to use aggregation type = max for a whole group instead of avg. For example: week temperature plot. Why max and not avg?
Thanks

**Comment by poblabs** [comment]
> Hello Pat.
> 
> > the closest aggregate_type to the observation will win
> 
> Yes, I've realized that, but my question is if it is correct to use aggregation type = max for a whole group instead of avg. For example: week temperature plot. Why max and not avg?
> Thanks

Oh I think I understand now. The [max is from the charts in 0.9.1 and older](https://github.com/poblabs/weewx-belchertown/blob/weewx-belchertown-0.9.1/bin/user/belchertown_highchartsSearchX.py#L314-L325) (which came from Gary, if I recall correctly). So the max now was just rolling that forward.
